cmd/src/batch_exec.go

```diff
@@ -174,6 +175,28 @@ func executeBatchSpecInWorkspaces(ctx context.Context, flags *executorModeFlags)
 		return errors.New("invalid execution, no steps to process")
 	}
 
+	// Pass the os.Environ to run steps to allow access to the secrets set
+	// in the executor environment.
+	// The executor runtime takes care of not forwarding any sensitive secrets
+	// from the host, so this is safe.
+	globalEnv := os.Environ()
+
+	// If the docker prefix registry is set, inject it into each step's container
+	// to fully qualify the container name
+	environ := map[string]string{}
+	for _, e := range globalEnv {
+		pair := strings.SplitN(e, "=", 2)
+		environ[pair[0]] = pair[1]
+	}
+
+	dockerPrefix, ok := environ["DOCKER_PREFIX_REGISTRY_URL"]
+	if ok {
+		for i := 0; i < len(task.Steps); i++ {
+			step := &task.Steps[i]
+			step.Container = fmt.Sprintf("%s/%s", dockerPrefix, step.Container)
```

Comment:
Changing the step might change the cache key, which which would case a mismatch between backend and src-cli 🤔 I'd need to validate that, but if that's the case we should only change it when we actually use the images, not on the step element itself.

Reply:
If you want to validate that yourself, you'd want to run a spec with 2 steps, execute, change the 2nd step command or whatever, and see if a re-execution would hit a cache result for step1 and then if src-cli correctly sees it and skips step 1 in execution.